src/general/HexEncoding.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+
+/// <summary>
+///   Simple implementation of hex encoding and decoding
+/// </summary>
+public static class HexEncoding
+{
+    /// <summary>
+    ///   Encodes a set of bytes as hex string in UTF8 format
+    /// </summary>
+    /// <param name="data">Where to read and put the result in</param>
+    /// <param name="length">How much of the data is to be used</param>
+    /// <param name="bytesWritten">How many bytes are written, on success should be equal to twice the length</param>
+    /// <exception cref="ArgumentException">If not enough space</exception>
+    public static void EncodeToHexInPlace(byte[] data, int length, out int bytesWritten)
+    {
+        if (length < 1)
+        {
+            bytesWritten = 0;
+            return;
+        }
+
+        if (data.Length < length * 2)
+            throw new ArgumentException("Hex encoding takes double the amount of space");
+
+        int outputWrite = length * 2 - 1;
+
+        for (int i = length - 1; i >= 0; --i)
+        {
+#if DEBUG
+            if (outputWrite <= i)
+                throw new Exception("Output write caught read point, this shouldn't happen");
+#endif
+
+            byte currentData = data[i];
+
+            // As we write in reverse order, we put the high nibble first
+            data[outputWrite--] = ByteToHexUtf8((byte)(currentData >> 4));
+            data[outputWrite--] = ByteToHexUtf8(currentData);
+        }
+
+        bytesWritten = length * 2;
+    }
+
+    /// <summary>
+    ///   Decodes hex encoded UTF8 text in place to raw bytes
+    /// </summary>
+    /// <param name="data">Input and the result</param>
+    /// <param name="length">How many bytes the input is</param>
+    /// <param name="outBytes">How many bytes have been written</param>
+    /// <returns>False if invalid characters are encountered (or input is empty)</returns>
+    public static bool DecodeFromHexInPlace(byte[] data, int length, out int outBytes)
+    {
```

Comment:
With prints, I determined that this function fails when reading a ':' symbol, probably at the start of the path.
It seems like this function isn't intended to read a full path, so maybe there should be a `startIndex` argument here?

Reply:
I found one problem preventing things from working on Linux and have now fixed that (there was one extra character that needed to be removed from the prefix). Could you re-test this on Windows as I don't have time today to do that myself?